src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -58,11 +58,13 @@ public partial class FunctionInvokingChatClient : DelegatingChatClient
     /// </summary>
     /// <param name="innerClient">The underlying <see cref="IChatClient"/>, or the next instance in a chain of clients.</param>
     /// <param name="logger">An <see cref="ILogger"/> to use for logging information about function invocation.</param>
-    public FunctionInvokingChatClient(IChatClient innerClient, ILogger? logger = null)
+    /// <param name="services">An optional <see cref="IServiceProvider"/> to use for resolving services required by the <see cref="AIFunction"/> instances being invoked.</param>
+    public FunctionInvokingChatClient(IChatClient innerClient, ILogger? logger = null, IServiceProvider? services = null)
         : base(innerClient)
     {
-        _logger = logger ?? NullLogger.Instance;
+        _logger = logger ?? (ILogger?)services?.GetService(typeof(ILogger<FunctionInvokingChatClient>)) ?? NullLogger.Instance;
```

Comment:
```suggestion
        _logger = logger ?? NullLogger.Instance;
```

Nit: falling back to using `GetService` for the logger is weird especially when you can provide the logger seperately. What if someone wanted to prevent `FunctionInvokingChatClient` from logging but still offer the service provider via arguments to their functions? It feels like an orthogonal concern. People could obviously adjust log levels for the appropriate category in their configuration, but what if they only want to disable logging for certain `FunctionInvokingChatClient` instances?

Also, it's odd for something that can be registered/resolved as a service to take a plain `ILogger` parameter. I see that [FunctionInvokingChatClientBuilderExtensions makes the appropriate `CreateLogger`](https://github.com/dotnet/extensions/blob/576fa221f57f80423a4852e446cee2aa2e937aab/src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClientBuilderExtensions.cs#L35) for you when wiring this up, but any service type that was registered directly with DI would need to take either an `ILogger<FunctionInvokingChatClient>` or `ILoggerFactory` for it to work.

Reply:
I don't think I've ever seen a method or constructor with both an `IServiceProvider` parameter and another parameter for a specific service that falls back to the service provider like this, so I wouldn't expect it just from the signature, but I can see it being convenient if you're manually newing up a `FunctionInvokingChatClient`. I'm fine with leaving the fallback there. You could always pass in `NullLogger.Instance` if you really don't like the fallback.

> We could change the latter to be ILoggerFactory if you think that's important.

On second thought. I think you can leave it as is given the design is never to register this type as a service directly. One upside of taking an `ILogger` over an `ILoggerFactory` is that people can pick their own category name if that care to.

tl;dr: leave everything as-is.